rdd/src/main/scala/eval/directive/RDDOpDirectives.scala

```diff
@@ -0,0 +1,99 @@
+package maml.rdd.eval.directive
+
+import maml.rdd.eval._
+import maml.rdd.ast._
+
+import maml.ast._
+import maml.eval._
+import maml.eval.directive._
+
+import geotrellis.raster._
+import geotrellis.spark._
+
+import org.apache.spark.rdd._
+
+import cats._
+import cats.implicits._
+import cats.data.{NonEmptyList => NEL, _}
+import Validated._
+
+
+object RDDOpDirectives {
+  private def doubleResults(grouped: Map[MamlKind, Seq[Result]]): Interpreted[List[Double]] =
+    grouped.getOrElse(MamlKind.Double, List.empty)
+      .map({ _.as[Double] })
+      .toList
+      .sequence
+
+  private def intResults(grouped: Map[MamlKind, Seq[Result]]): Interpreted[List[Int]] =
+    grouped.getOrElse(MamlKind.Int, List.empty)
+      .map({ _.as[Int] })
+      .toList
+      .sequence
+
+  private def spatialRDDResults(grouped: Map[MamlKind, Seq[Result]]): Interpreted[List[TileLayerRDD[SpatialKey]]] =
+      grouped(MamlKind.Tile)
+        .map({ _.as[RDD[(SpatialKey, Tile)] with Metadata[TileLayerMetadata[SpatialKey]]] })
+        .toList
+        .sequence
+
+  val additionDirective = Directive { case (a@Addition(_), childResults) =>
+    val grouped = childResults.groupBy(_.kind)
+
+    val scalarSums =
+      (doubleResults(grouped) |@| intResults(grouped)).map { case (dbls, ints) =>
+        dbls.sum + ints.sum
+      }
+
+      (spatialRDDResults(grouped) |@| scalarSums).map({ case (rdds, sums) =>
+        val rddSums = rdds.reduce({
+          (rdd1: TileLayerRDD[SpatialKey], rdd2: TileLayerRDD[SpatialKey]) =>
+            rdd1.withContext { _ localAdd rdd2 }
+        })
+      SpatialRDDResult(rddSums.withContext { _ localAdd sums })
+      })
+  }
+
+  val subtractionDirective = Directive { case (a@Subtraction(_), childResults) =>
+    val results = childResults.reduce({ (res1: Result, res2: Result) =>
+      (res1, res2) match {
+        case (SpatialRDDResult(rdd1), SpatialRDDResult(rdd2)) => SpatialRDDResult(rdd1.withContext{ _ - rdd2 })
```

Comment:
what if `rdd1` and `rdd2` are not aligned, their `LayoutDefinition` is not matched or they are in difference CRS? There should be an error based on that.